### PR TITLE
Remove Python 3.8 Support from Apache Beam

### PR DESCRIPTION
**Pull request by jrmccluskey**
Removes Python 3.8 support from Apache Beam and cleans up a number of references to the version. Also stops building 3.8 snapshot containers. 

DO NOT MERGE while 2.60 release is in-flight

Part of #31192

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by jrmccluskey** [comment]
R: @tvalentyn @damccorm 

Since this also impacts things like the build wheels workflow this should be held until the release is further along

File: sdks/python/apache_beam/__init__.py [hunk 1]
```diff
@@ -75,12 +75,6 @@
         'This version of Apache Beam has not been sufficiently tested on '
         'Python %s.%s. You may encounter bugs or missing features.' %
         (sys.version_info.major, sys.version_info.minor))
-  elif sys.version_info.minor == 8:
```
Comment (damccorm) [inline sdks/python/apache_beam/__init__.py:78]: We should also change the boundaries of the above condition to include 3.8

**Review by damccorm — APPROVED** [review]
Sounds good - lets leave it until the release is finalized

File: playground/infrastructure/cloudbuild/playground_cd_examples.sh [hunk 2]
```diff
@@ -97,15 +97,15 @@ LogOutput "Installing python and dependencies."
 export DEBIAN_FRONTEND=noninteractive
 apt install -y apt-transport-https ca-certificates software-properties-common curl unzip apt-utils > /dev/null 2>&1
 add-apt-repository -y ppa:deadsnakes/ppa > /dev/null 2>&1 && apt update > /dev/null 2>&1
-apt install -y python3.8 python3.8-distutils python3-pip > /dev/null 2>&1
-apt install -y --reinstall python3.8-distutils > /dev/null 2>&1
+apt install -y python3.9 python3.9-distutils python3-pip > /dev/null 2>&1
```
Comment (tvalentyn) [inline playground/infrastructure/cloudbuild/playground_cd_examples.sh:100]: FYI, I don't know if python3.9-distutils is available (https://github.com/deadsnakes/issues/issues/150), and not sure if this package is still necessary.

Comment (tvalentyn) [reply playground/infrastructure/cloudbuild/playground_cd_examples.sh:100]: is this code exercised by some IT suite?

File: sdks/python/apache_beam/__init__.py (hunk 1 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/__init__.py:78]: +1

File: sdks/python/setup.py [hunk 3]
```diff
@@ -155,7 +155,7 @@ def cythonize(*args, **kwargs):
 # Exclude 1.5.0 and 1.5.1 because of
 # https://github.com/pandas-dev/pandas/issues/45725
 dataframe_dependency = [
-    'pandas>=1.4.3,!=1.5.0,!=1.5.1,<2.3;python_version>="3.8"',
+    'pandas>=1.4.3,!=1.5.0,!=1.5.1,<2.3;python_version>="3.9"',
```
Comment (tvalentyn) [inline sdks/python/setup.py:158]: we can omit python_version>="3.9" since now it includes all the versions we support

Comment (tvalentyn) [reply sdks/python/setup.py:158]: conditions like this make sense when we install different dependency versions depending on Python version, for example, pandas 1.x for Python 3.7, and Pandas 2.x for all other versions.

File: sdks/python/setup.py [hunk 4]
```diff
@@ -271,18 +271,13 @@ def get_portability_package_data():
   return files
 
 
-python_requires = '>=3.8'
+python_requires = '>=3.9'
```
Comment (tvalentyn) [inline sdks/python/setup.py:274]: (drive by): line 276 should have `>= 13` or `> 12`

**Review by tvalentyn — APPROVED** [review]
LGTM otherwise

File: playground/infrastructure/cloudbuild/playground_cd_examples.sh (hunk 2 above)
Comment (jrmccluskey) [reply playground/infrastructure/cloudbuild/playground_cd_examples.sh:100]: I believe these scripts are invoked for playground deployment, not exercised by a test. Looks like we can get away with python3-distutils

File: sdks/python/setup.py (hunk 4 above)
Comment (jrmccluskey) [reply sdks/python/setup.py:274]: fixed

File: sdks/python/setup.py (hunk 3 above)
Comment (jrmccluskey) [reply sdks/python/setup.py]: fixed

File: sdks/python/apache_beam/__init__.py (hunk 1 above)
Comment (jrmccluskey) [reply sdks/python/apache_beam/__init__.py:78]: fixed

**Comment by jrmccluskey** [comment]
The MacOS failures are due to #32811, unrelated to this PR

**Pull request merged**